tests/file/FileTest.php

```diff
@@ -64,13 +64,28 @@ final class FileTest extends HackTest {
     $dir = sys_get_temp_dir().'/hsl-test-'.PseudoRandom\int(0, 99999999);
     mkdir($dir);
     using ($tf = File\temporary_file($dir.'/foo')) {
-      expect(
-        Str\starts_with($tf->getHandle()->getPath(), $dir.'/foo'),
-      )
+      expect(Str\starts_with($tf->getHandle()->getPath(), $dir.'/foo'))
         ->toBeTrue();
     }
   }
 
+  public async function testLeakyTemporaryFile(): Awaitable<void> {
+    $tf1 = File\leaky_temporary_file();
+    expect($tf1->getPath())->toNotContainSubstring('XXXXXX');
+
+    $tf2 = File\leaky_temporary_file();
+    expect($tf1->getPath())->toNotEqual($tf2->getPath());
+    expect(\file_exists($tf1->getPath()))->toBeTrue();
+    $tf1->close();
+    expect(\file_exists($tf1->getPath()))->toBeTrue();
+    await $tf2->writeAllAsync('hello');
+    $tf2->close();
+    expect(\file_get_contents($tf2->getPath()))->toEqual('hello');
```

Comment:
Maybe annotate this line to say that the file should be leaked here?

Otherwise it's kinda unclear what this is testing